### PR TITLE
implement StartMigration rpc

### PR DESCRIPTION
**Pull request by songgao**
Related: https://github.com/keybase/kbfs-server/pull/462

Also I'm realizing I haven't done anything in mdserver_remote.go to support iTeam migration yet. I'll create a ticket and work  on it later.

### CODE REVIEW

File: libkbfs/keybase_service_base.go [hunk 1]
```diff
@@ -1040,6 +1041,22 @@ func (k *KeybaseServiceBase) TeamDeleted(ctx context.Context,
...
 
+// StartMigration implements keybase1.ImplicitTeamMigrationInterface for
+// KeybaseServiceBase.
+func (k *KeybaseServiceBase) StartMigration(ctx context.Context,
+	folder keybase1.Folder) (err error) {
+	mdServer := k.config.MDServer()
+	if mdServer == nil {
+		return errors.New("no mdserver")
+	}
+	fav := NewFavoriteFromFolder(folder)
```
Comment (strib) [inline libkbfs/keybase_service_base.go]: Why make a favorite?  You can just use `folder.Name` and `folder.Type` in the call below directly.

File: libkbfs/mdserver_memory.go [hunk 2]
```diff
@@ -88,6 +88,7 @@ type mdServerMemShared struct {
 	// tracks expire time and holder
 	lockIDs              map[mdLockMemKey]mdLockMemVal
 	implicitTeamsEnabled bool
+	iTeamMigrationLocks  map[tlf.ID]struct{}
```
Comment (strib) [inline libkbfs/mdserver_memory.go]: Slightly prefer a `map[tlf.ID]bool` instead so you can just `return md.iTeamMigrationLocks[id]` below.

File: libkbfs/mdserver_memory.go [hunk 3]
```diff
@@ -465,6 +467,19 @@ func (md *MDServerMemory) GetRange(ctx context.Context, id tlf.ID,
 	}
 }
 
+func (md *MDServerMemory) iTeamMigrationRemoveLock(id tlf.ID) {
```
Comment (strib) [inline libkbfs/mdserver_memory.go]: I'm not sure, but it seems likely that these two functions will be called with `md.lock` already held.  But we can fix that when that code is written, I guess.

File: libkbfs/keybase_service_base.go (hunk 1 above)
Comment (songgao) [reply libkbfs/keybase_service_base.go]: It's to reuse `keybase1.FolderType`->`tlf.Type` conversion. Would you prefer if I do a switch here?

File: libkbfs/mdserver_memory.go (hunk 2 above)
Comment (songgao) [reply libkbfs/mdserver_memory.go]: Hmmm OK!

File: libkbfs/mdserver_memory.go (hunk 3 above)
Comment (songgao) [reply libkbfs/mdserver_memory.go]: Yeah; good point. I'll fix the in the follow up ticket which will port the migration checking into `MDServerMemory`

File: libkbfs/keybase_service_base.go (hunk 1 above)
Comment (strib) [reply libkbfs/keybase_service_base.go]: Ah I see. Ok.  I guess that makes sense.  Can you please add a comment about it though?

File: libkbfs/interfaces.go [hunk 4]
```diff
@@ -1388,6 +1388,11 @@ type MDServer interface {
 	// expired, this is a no-op.
 	ReleaseLock(ctx context.Context, tlfID tlf.ID, lockID keybase1.LockID) error
 
+	// StartImplicitTeamMigration tells mdserver to put a implicit team
+	// migration lock on id, which prevents any rekey MD writes from going
```
[the remark this answers is not in the document]
Comment (songgao) [reply libkbfs/interfaces.go]: Those still go in. I'll update the comment to mention it.

**Pull request merged**